base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt

```diff
@@ -22,59 +23,183 @@ import androidx.annotation.DrawableRes
 import androidx.annotation.IdRes
 import androidx.annotation.LayoutRes
 import androidx.annotation.StyleRes
+import androidx.appcompat.content.res.AppCompatResources
 import androidx.core.content.ContextCompat
 
 class PopupTooltip(
-        //Required parameters
-        private var anchorView: View,
-        private var message: CharSequence,
-        @LayoutRes private var layout: Int,
-        @IdRes private var textViewId: Int,
-        @IdRes private var tooltipBackgroundId: Int,
-        //Tooltip layout
-        private var elevation: Int = DEFAULT_ELEVATION,
-        private var popupWidth: Int = TOOLTIP_WIDTH,
-        //Tooltip behaviour
-        private var indefinite: Boolean = false,
-        private var showDuration: Long = 3500L,
-        private var verticalMargin: Int = 0,
-        private var horizontalMargin: Int = 0,
-        private var horizontalMarginClipped: Int = 0,
-        private var verticalMarginClipped: Int = 0,
-        private var insideTouchDismissible: Boolean = true,
-        private var outsideTouchDismissible: Boolean = true,
-        private var respectScreenBounds: Boolean = false,
-        private var modal: Boolean = true,
-        @StyleRes private var animation: Int? = null,
-        //Tooltip arrow
-        private var arrowWidth: Int = ARROW_WIDTH,
-        private var arrowHeight: Int = ARROW_HEIGHT,
-        private var arrow: Boolean = false,
-        @DrawableRes private var arrowDrawable: Int? = null,
-        @ColorRes private var arrowStrokeColor: Int = android.R.color.white,
-        @IdRes private var arrowBackgroundId: Int? = null,
-        @IdRes private var arrowStrokeId: Int? = null
-) : PopupWindow() {
+    /**
+     * The view to anchor the tooltip to.
+     */
```

Comment:
This types of comments makes this class really hard to read. Can we take some time to improve it a little bit?

One way would be to inline the comment whenever possible like this

```
/** The view to anchor the tooltip to. */
private val anchorView: View,

/** The message to display on the tooltip. */
private val message: CharSequence,

/** The layout resource that contains the tooltip..layout. */
@LayoutRes private val layoutRes: Int,

...
```

Or even creating a comment for the whole constructor on the top

```
/**
 * @constructor Primary constructor for the tooltip view.
 * @property anchorView The view to anchor the tooltip to.
 * @property message The message to display on the tooltip.
 * @property ...
 */ 
```

Reply:
changed 👍